operator/pkg/metrics/server.go

```diff
@@ -0,0 +1,211 @@
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+	"unicode"
+
+	"github.com/benbjohnson/clock"
+	"github.com/gorilla/mux"
+	metricsv1alpha1 "github.com/keptn/lifecycle-toolkit/operator/apis/metrics/v1alpha1"
+	"github.com/open-feature/go-sdk/pkg/openfeature"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+type Metrics struct {
+	gauges map[string]prometheus.Gauge
+}
+
+var metrics Metrics
+
+var instance *serverManager
+var smOnce sync.Once
+
+type serverManager struct {
+	server        *http.Server
+	ticker        *clock.Ticker
+	ofClient      *openfeature.Client
+	exposeMetrics bool
+}
+
+func StartServerManager(ctx context.Context, exposeMetrics bool) {
+	smOnce.Do(func() {
+		metrics.gauges = make(map[string]prometheus.Gauge)
+		instance = &serverManager{
+			ticker:        clock.New().Ticker(10 * time.Second),
+			ofClient:      openfeature.NewClient("klt"),
+			exposeMetrics: exposeMetrics,
+		}
+		instance.Start(ctx)
+	})
+}
+
+func (m *serverManager) Start(ctx context.Context) {
```

Comment:
let's make this function unexported since it should only be called from within the exported `StartServerManager` function